src/App.js

```diff
@@ -1,27 +1,46 @@
-// src/App.js
 import React, { useState } from 'react';
+import { BrowserRouter as Router, Route, Routes, Link } from 'react-router-dom';
 import StockList from './components/StockList';
 import StockPrice from './components/StockPrice';
+import News from './components/News';
+import SaveStock from './components/SaveStock';
+import StockPieChart from './components/StockPieChart';
 import './App.css';
 
 const App = () => {
   const [selectedSymbol, setSelectedSymbol] = useState('');
-  const stockSymbols = ['AAPL', 'GOOGL', 'MSFT', 'AMZN', 'TSLA']; // Example stock symbols
 
   return (
-    <div className="App">
-      <div className="top-bar">
-        <h1>Stock Price Checker</h1>
-      </div>
-      <div className="content">
-        <div className="left-pane">
-          <StockList symbols={stockSymbols} onSelect={setSelectedSymbol} />
+    <Router>
+      <div className="App">
+        <div className="top-bar">
+          <h1>Stock Price Checker</h1>
+          <nav>
+            <Link to="/">Home</Link>
+            <Link to="/news">News</Link>
+            <Link to="/save-stock">Save Stock</Link>
+            <Link to="/pie-chart">Pie Chart</Link>
+          </nav>
         </div>
-        <div className="main-pane">
-          <StockPrice selectedSymbol={selectedSymbol} />
+        <div className="content">
+          <Routes>
+            <Route path="/" element={
+              <>
+                <div className="left-pane">
+                  <StockList onSelect={setSelectedSymbol} />
+                </div>
+                <div className="main-pane">
+                  <StockPrice selectedSymbol={selectedSymbol} />
```

Comment:
It might not be a bad idea to hide the `StockPrice` when no stocks are selected. I would say this is a low priority change though.